instrumentation/spring/spring-boot-autoconfigure/build.gradle.kts

```diff
@@ -37,7 +37,7 @@ dependencies {
   library("org.springframework.boot:spring-boot-starter-web:$springBootVersion")
   library("org.springframework.boot:spring-boot-starter-webflux:$springBootVersion")
 
-  compileOnly("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi")
+  implementation("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure")
```

Comment:
It should not be `library("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure")`, as the Spring dependencies? cc @laurit who should know

Reply:
`library` means `compileOnly` + `testImplementation`. Using `compileOnly` is important in `javaagent` modules, which this module is not. It is also used to support the latest dep test where we change the version for `library` dependencies to latest version. In my opinion here `implementation` is the correct choice as it is a required dependency for this module.